app/_src/gateway/plugin-development/quickstart.md

```diff
@@ -0,0 +1,679 @@
+
+    Once the test file from the previous step is saved in the `spec/my-plugin/01-integration_spec.lua` file, 
+    execute a test run:
+
+    ```sh
+    pongo run
+    ```
+
+    You should see a successful report that looks similar to the following:
+
+    ```sh
+    [pongo-INFO] auto-starting the test environment, use the 'pongo down' action to stop it
+    Kong version: 3.6.1
+    
+    [==========] Running tests from scanned files.
+    [----------] Global test environment setup.
+    [----------] Running tests from /kong-plugin/spec/my-plugin/01-integration_spec.lua
+    [----------] Running tests from /kong-plugin/spec/my-plugin/01-integration_spec.lua
+    [ RUN      ] /kong-plugin/spec/my-plugin/01-integration_spec.lua:63: my-plugin: [#postgres] The response gets a 'X-MyPlugin' header
+    [       OK ] /kong-plugin/spec/my-plugin/01-integration_spec.lua:63: my-plugin: [#postgres] The response gets a 'X-MyPlugin' header (6.59 ms)
+    [ RUN      ] /kong-plugin/spec/my-plugin/01-integration_spec.lua:63: my-plugin: [#off] The response gets a 'X-MyPlugin' header
+    [       OK ] /kong-plugin/spec/my-plugin/01-integration_spec.lua:63: my-plugin: [#off] The response gets a 'X-MyPlugin' header (4.76 ms)
+    [----------] 2 tests from /kong-plugin/spec/my-plugin/01-integration_spec.lua (23022.12 ms total)
+    
+    [----------] Global test environment teardown.
+    [==========] 2 tests from 1 test file ran. (23022.80 ms total)
+    [  PASSED  ] 2 tests.
+    ```
+
+    Pongo can also run as part of a Continuous Integration (CI) system, the 
+    [repository documentation](https://github.com/Kong/kong-pongo?tab=readme-ov-file#setting-up-ci) has more details. 
+
+    Next, let's see how we can modify the behavior of our plugin with configurable values.
+
+1. **Make `my-plugin` configurable**
+
+    Plugins are not typically as simplistic as our current example. You will likely wish
+    to control the behavior of your plugin using configurable values. 
+    Let's add some configuration fields to our `schema.lua` file allowing us to 
+    make the plugin behavior configurable without having to redeploy or restart.
+
+    {{site.base_gateway}} provides a module of 
+    base [type definitions](https://github.com/Kong/kong/blob/master/kong/db/schema/typedefs.lua) that can
+    help us with common types related to API gateway plugins. 
+
+    At the top of the `schema.lua` file we include the {{site.base_gateway}} 
+    `typedefs` module with the statement:
+
+    ```lua
+    local typedefs = require "kong.db.schema.typedefs"
+    ```
+
+    Below is a Lua snippet showing a typical 
+    [configuration field](gateway/{{site.page_version}}/plugin-development/configuration/#describing-your-configuration-schema)
+    that defines a field named `response_header_name`. 
+
+    Here we use the `header_name` type definition which defines the field to be a string that cannot be null and 
+    conforms to the gateway rules for header names.
```

Comment:
Is it worth showing the contents of the typedef to explain that a typedef is a reusable schema definition?

Reply:
Yes, I think so. I will do this after I refactor for sub-chapters so the length stays under control